### PR TITLE
OSDOCS-3539: add maxConnections parameter

### PR DESCRIPTION
**Pull request by jldohmann**
Version(s): 4.11+

Issue: https://issues.redhat.com/browse/OSDOCS-3539

Link to docs preview:
- https://deploy-preview-45265--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-parameters_configuring-ingress
- https://deploy-preview-45265--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-setting-max-connections_configuring-ingress

Additional information:

### CODE REVIEW

**Review by jldohmann — COMMENTED** [review]
@frobware this is the doc for[ NE-577](https://issues.redhat.com/browse/NE-577), could you PTAL for SME feedback/ack? Thank you 🙇

I left one clarification question I have inline

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 1]
```diff
@@ -219,6 +219,19 @@ supports up to `64` threads. If this field is empty, the Ingress Controller uses
 
 * `tunnelTimeout` specifies how long a tunnel connection, including websockets, remains open while the tunnel is idle. If unset, the default timeout is `1h`.
 
+* `maxConnections` specifies the maximum number of simultaneous connections that can be established per HAProxy process. Increasing this value lets each ingress controller pod handle more connections at the cost of additional system resources. Permitted values are `0`, `-1`, any value within the range `2000` and `2000000`, or the field can be left empty. 
+
+** If this field is left empty or has the value `0`, the ingress controller will use the default value of `20000`.
```
Comment (frobware) [inline modules/nw-ingress-controller-configuration-parameters.adoc:224]: We should note that the default of 20,000 is subject to change in future releases.

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 2]
```diff
@@ -219,6 +219,19 @@ supports up to `64` threads. If this field is empty, the Ingress Controller uses
...
+
+** If this field is left empty or has the value `0`, the ingress controller will use the default value of `20000`. 
+
+** If the field has the value of `-1`, then HAProxy will dynamically compute a maximum value based on the available `ulimits` in the running container. This process results in a large computed value, and therefore each HAProxy process will incur significant memory usage compared to the current default value of `20000`.
+
+** If the field has a value that is greater than the current operating system limit, the HAProxy process will not start.
+
+** If you choose a discrete value and the router pod is migrated to a new node, it is possible the new node does not have an identical `ulimit` configured. In such cases, the pod fails to start.
+
+** If you have pods with different `ulimits` configured, and you choose a discrete value, it is recommended to use the value of `-1` for this field so that the value is calculated at runtime.
```
[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/nw-ingress-controller-configuration-parameters.adoc:232]: @frobware thanks, would it make sense to say that this second mention of `value` refers to the `maximum number of connections calculated at runtime`? Just trying to remove vagueness in wording

**Comment by jldohmann** [comment]
d1e3169..ef36fb2 implement initial feedback

File: modules/nw-ingress-controller-configuration-parameters.adoc (hunk 1 above)
Comment (jldohmann) [reply modules/nw-ingress-controller-configuration-parameters.adoc]: thanks, added

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 3]
```diff
@@ -219,6 +219,19 @@ supports up to `64` threads. If this field is empty, the Ingress Controller uses
 
 * `tunnelTimeout` specifies how long a tunnel connection, including websockets, remains open while the tunnel is idle. If unset, the default timeout is `1h`.
 
+* `maxConnections` specifies the maximum number of simultaneous connections that can be established per HAProxy process. Increasing this value lets each ingress controller pod handle more connections at the cost of additional system resources. Permitted values are `0`, `-1`, any value within the range `2000` and `2000000`, or the field can be left empty. 
+
+** If this field is left empty or has the value `0`, the ingress controller will use the default value of `20000`. 
+
+** If the field has the value of `-1`, then HAProxy will dynamically compute a maximum value based on the available `ulimits` in the running container. This process results in a large computed value, and therefore each HAProxy process will incur significant memory usage compared to the current default value of `20000`.
```
[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/nw-ingress-controller-configuration-parameters.adoc]: thanks for clarifying, fixed

**Comment by jldohmann** [comment]
@frobware I've implemented your feedback, could you PTAL again for SME ack? thank you 😃

**Comment by jldohmann** [comment]
@frobware I've implemented your feedback, could you PTAL again for SME ack? thank you 😃

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 4]
```diff
@@ -216,6 +216,19 @@ supports up to `64` threads. If this field is empty, the Ingress Controller uses
...
+
+** If this field is left empty or has the value `0`, the ingress controller will use the default value of `20000`. This value is subject to change in future releases.
+
+** If the field has the value of `-1`, then HAProxy will dynamically compute a maximum value based on the available `ulimits` in the running container. This process results in a large computed value that will incur significant memory usage compared to the current default value of `20000`.
+
+** If the field has a value that is greater than the current operating system limit, the HAProxy process will not start.
+
+** If you choose a discrete value and the router pod is migrated to a new node, it is possible the new node does not have an identical `ulimit` configured. In such cases, the pod fails to start.
+
+** If you have pods with different `ulimits` configured, and you choose a discrete value, it is recommended to use the value of `-1` for this field so that the maximum number of connections is calculated at runtime.
```
[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/nw-ingress-controller-configuration-parameters.adoc]: Thank you! Fixed

**Comment by jldohmann** [comment]
Thanks @frobware I've implemented your feedback.

@ShudiLi could you PTAL for QA feedback/ack? Thank you!

**Comment by ShudiLi** [comment]
/label qe-approved
@jldohmann thanks

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 5]
```diff
@@ -216,6 +216,19 @@ supports up to `64` threads. If this field is empty, the Ingress Controller uses
 
 * `tunnelTimeout` specifies how long a tunnel connection, including websockets, remains open while the tunnel is idle. If unset, the default timeout is `1h`.
 
+* `maxConnections` specifies the maximum number of simultaneous connections that can be established per HAProxy process. Increasing this value lets each ingress controller pod handle more connections at the cost of additional system resources. Permitted values are `0`, `-1`, any value within the range `2000` and `2000000`, or the field can be left empty.
```
Comment (ahardin-rh) [inline modules/nw-ingress-controller-configuration-parameters.adoc:219]: Just a suggestion:
"allows" vs "lets"

**Comment by ahardin-rh** [comment]
Looks great! Only one minor suggestion, which you can take or leave. :smile:

**Comment by jldohmann** [comment]
3208569..03be3bc implement minor docs suggestion

**Pull request merged**